shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/NotificationService.java

```diff
@@ -0,0 +1,20 @@
+package com.redhat.service.bridge.shard.operator;
+
+import com.redhat.service.bridge.infra.models.dto.BridgeDTO;
+import com.redhat.service.bridge.infra.models.dto.ProcessorDTO;
+import com.redhat.service.bridge.shard.operator.metrics.ManagerRequestType;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpResponse;
+
+public interface NotificationService {
+    Uni<HttpResponse<Buffer>> notifyBridgeStatusChange(BridgeDTO bridgeDTO);
+
+    Uni<HttpResponse<Buffer>> notifyProcessorStatusChange(ProcessorDTO processorDTO);
+
+    void updateManagerRequestMetricsOnSuccess(ManagerRequestType requestType, HttpResponse<Buffer> successResponse);
+
+    void updateManagerRequestMetricsOnFailure(ManagerRequestType requestType, Throwable error);
+
+}
```

Comment:
`updateManagerRequestMetricsOnSuccess` is not really related to notifications. I'd create a `ManagerClient` instead of this `NotificationService` that expose the `notifyBridgeStatusChange`, `notifyProcessorStatusChange`, `getBridgesToDeployOrDelete`, `getProcessorsToDeployOrDelete` implemented as

```
return getAuthenticatedRequest(webClientManager.get(APIConstants.SHARD_API_BASE_PATH + "processors"), HttpRequest::send)
                .onItem().invoke(success -> notificationService.updateManagerRequestMetricsOnSuccess(ManagerRequestType.FETCH, success))
                .onFailure().invoke(failure -> notificationService.updateManagerRequestMetricsOnFailure(ManagerRequestType.FETCH, failure))
```
and same for the bridges. 

Note that currently you had to duplicate the `getAuthenticatedRequest` in the `NotificationServiceImpl` and `ManagerSyncServiceImpl` which is not ideal. If you go for a service that aims to handle all the communications with the manager you can better follow the single responsibility principle